src/esp/assets/ResourceManager.h

```diff
@@ -1299,6 +1295,24 @@ class ResourceManager {
   Corrade::Containers::Pointer<Importer> fileImporter_;
 
   // ======== Physical parameter data ========
+
+  // /**
+  //  * @brief Manages all construction and access to asset attributes.
+  //  */
+  // managers::AssetAttributesManager::ptr assetAttributesManager_ = nullptr;
+  // /**
+  //  * @brief Manages all construction and access to object attributes.
+  //  */
+  // managers::ObjectAttributesManager::ptr objectAttributesManager_ = nullptr;
+  /**
+   * @brief Manages all construction and access to physics world attributes.
+   */
+  managers::PhysicsAttributesManager::ptr physicsAttributesManager_ = nullptr;
```

Comment:
Is there any case that we must share the pointer?
If not, I strongly recommend to use unique pointer, which should be our default choice for smart pointers.

Reply:
yes, we share it by giving it to the user, so they can access it directly.  that is the purpose of the PR, instead of having them go through Resource Manager, they will get the object and manipulate the attributes themselves.

Reply:
Do they get ownership?  I am not sure if this has any OpenGL resources, but if it does, then the user shouldn't get ownership (we assume all ownership of OpenGL).

Reply:
These are just attributes managers - they only manage the objects that describe the configurations of various constructs, like objects, primitive assets, scenes and the physics world.  They do not actually touch on assets, that is still the sole purview of ResourceManager.

Reply:
You may need to bind the subclass into python for things to work correctly.  We use virtual methods here: https://github.com/facebookresearch/habitat-sim/blob/master/src/esp/bindings/SceneBindings.cpp#L143